### ISSUE TITLE
query loop

### ISSUE THREAD

**Issue by tc232** [Ashish-Github193/facebook-ads-library#3]
Hi,

I've been at it for a few hours and I keep getting looped when inputting a search query for a target word.
![Screenshot 2023-08-25 212816](https://github.com/Ashish-Github193/facebook-ads-library/assets/104802465/5bce8141-e939-4bad-99f4-d9e31100f134)

Thank you

**Comment by Ashish-Github193** [comment]
Hi,

Could you please provide me with the following details? This will help me understand what actually went wrong:

1. Did you run it multiple times?
2. Did the code print anything in the console while being in a loop?
3. Do you have Chrome installed on your system?
Thanks.

**Comment by tc232** [comment]
I was able to successfully run it however I am running into the issue that it only scraped the first page and does not continue down. Is that correct?

**Comment by Ashish-Github193** [comment]
No, it should have scraped all the pages. There might have been an error. Could you please verify this by commenting out the 'Headless mode' in the initialization of the Selenium utility function?

**Comment by tc232** [comment]
Hello,

When the chrome browser opens it scrapes today until it comes to a point where the page needs to load more ads. I think it believes that is the end of the ad count and shuts down.

**Comment by havardl** [comment]
Same happens when I try to run it. Seems like it's not able to find the main_division xpath, and then just times out. Perhaps some change in the source code on the website?

**Comment by Ashish-Github193** [comment]
Thanks, I should create a config file to store all the xpaths.

**Comment by havardl** [comment]
Sorry, I mean it seems like the xpath config which is set there now breaks, as the current source code of the webpage might have changed.

**Comment by Ashish-Github193** [comment]
Yes, I understand your point. What I meant is that if I put all the XPaths in a configuration file, it will make it easier for everyone to quickly modify them in case something breaks, allowing us to continue smoothly.

**Comment by joaovytor123** [comment]
Im stucked in the same query loop, when try to search for a query the process finished without saving the ads.
